File: src/ts/minimumLengthEncoding.ts
export function minimumLengthEncoding (words: string[]): number {
  const filter = new Set<string>(words)
  for (const word of words) {
    for (let i = 1; i < word.length; i += 1) {
      filter.delete(word.substring(i))
    }
  }

  let result = 0
  for (const word of filter) {
    result += word.length + 1
  }

  return result
}

File: src/rs/minimum_length_encoding.rs
use std::collections::HashSet;

#[allow(dead_code)]
pub struct MinimumLengthEncoding {}

#[allow(dead_code)]
impl MinimumLengthEncoding {
    pub fn minimum_length_encoding(words: Vec<String>) -> i32 {
        let mut filter = HashSet::<String>::new();

        for word in &words {
            filter.insert(word.clone());
        }

        for word in &words {
            for i in 1..word.len() {
                let sub = &word[i..];
                filter.remove(sub);
            }
        }

        let mut result = 0usize;
        for word in filter {
            result += word.len() + 1;
        }

        result as i32
    }
}
